src/currency_lib_calls.c

```diff
@@ -6,13 +6,41 @@
 
 #include "currency_lib_calls.h"
 
-int get_printable_amount(buf_t *coin_config,
-                         char *application_name,
-                         unsigned char *amount,
-                         unsigned char amount_size,
-                         char *printable_amount,
-                         unsigned char printable_amount_size,
-                         bool is_fee) {
+/*
+  Inter-application doc:
+
+  `os_lib_call` is called with an `unsigned int [5]` argument.
+  The first argument is used to choose the app to start, the rest is used as
+  this app's main arguments.
+  For instance, in Ethereum, the args are casted into a struct like:
+
+  ```
+  struct libargs_s {
+    unsigned int id;
+    unsigned int command;
+    chain_config_t *chain_config;
+    union {
+      check_address_parameters_t *check_address;
+      create_transaction_parameters_t *create_transaction;
+      get_printable_amount_parameters_t *get_printable_amount;
+    };
+  };
+  ```
+  ... so, in the following functions, the mapping is:
+
+  `libcall_params[1]` -> `libargs_s.id`
+  `libcall_params[2]` -> `libargs_s.command`
+  `libcall_params[3]` -> `libargs_s.chain_config`
+  `libcall_params[4]` -> `libargs_s` union
+ */
+
+int get_printable_amount(const buf_t *const coin_config,
+                         const char *const application_name,
+                         const unsigned char *const amount,
+                         const unsigned char amount_size,
+                         char *const printable_amount,
```

Comment:
`printable_amount` is the field used to store the result of the coin application: there is a `strncpy(printable_amount, ..., ...)` after the  `os_lib_call`, so it can't be `const`